crates/runtime/src/embeddings/connector.rs

```diff
@@ -231,14 +236,34 @@ impl EmbeddingConnector {
             .collect();
         let mut provider = IndexedTableProvider::new(Arc::clone(&inner_table_provider));
         for (column, config) in embedding_columns {
+            let (dataset_columns, index_schema) =
+                if config.chunking.as_ref().is_some_and(|cfg| cfg.enabled) {
+                    Self::updated_chunked_search_index_format(
+                        &inner_table_provider,
+                        dataset,
+                        column.clone(),
+                    )
+                } else {
+                    (dataset.columns.clone(), inner_table_provider.schema())
+                };
+
             let vector_index = super::index::s3::try_from_dataset(
                 &dataset.name,
                 column,
                 config.clone(),
                 vector_store,
-                Arc::clone(&inner_table_provider),
+                // Primary key. Use override from spicepod, fallback to underlying [`TableProvider`].
+                get_primary_keys(&inner_table_provider)
+                    .await
+                    .boxed()
+                    .map_err(|e| DataConnectorError::UnableToConnectInternal {
```

Comment:
This is the error message that @peasee said we should improve on.

Reply:
Not here, but I can make that change separately